feat(bedrock): add requestId logging to BedrockModel

sanjanasekar opened the pull request:
Implements bedrock-request-id-logging spec requirements

Adds request ID logging for Bedrock API calls using a dedicated child logger (strands.models.bedrock.requestId). This implements the approach suggested by @zastrowm - using Python's standard logging hierarchy instead of a custom configuration parameter.

Key changes:

Log request ID at DEBUG level for successful streaming/non-streaming requests
Log request ID at INFO level for failed requests (when users most need it for troubleshooting)
Child logger inherits from parent logger, respecting existing log configuration
Usage

Users can enable request ID logging via Python's standard logging configuration:
```
  import logging

 # Option 1: Enable via the child logger directly
  logging.getLogger('strands.models.bedrock.requestId').setLevel(logging.DEBUG)

 # Option 2: Enable via parent logger (will also show other debug logs)
  logging.getLogger('strands.models.bedrock').setLevel(logging.DEBUG)
```

dbschmigelski commented on src/strands/models/bedrock.py:
```diff
@@ -681,6 +681,8 @@ def _stream(
             if streaming:
                 response = self.client.converse_stream(**request)
+                request_id = response.get('ResponseMetadata', {}).get('RequestId')
+                logger.info(f"Bedrock converse_stream requestId: {request_id}")
```
Hi, I proposed some changes here that I want to discuss with the team. We can introduce a logging configuration field to enable this. I do NOT want to place this on the base model interface just yet but we have the ability to do that in the future.

zastrowm commented on src/strands/models/bedrock.py:
```diff
@@ -169,6 +186,8 @@ def __init__(
             region_name=resolved_region,
         )
 
+        self.logging_config: _ModelLoggingConfig = {**_DEFAULT_MODEL_LOGGING_CONFIG, **(logging_config or {})}
```
Rather than going down this path, could/should we use a different logger for this? That way the caller could configure logging of the request id using python instead of us needing to expose options for each thing we want to be configurable.

E.g. this could be:

```python
logger_requests = logging.getLogger(__name__ + "::requestId")
```

dbschmigelski replied:
I like this because we are more hands off.

I'm going to consider the pros and cons of this compared to env vars.

What I'm concerned about for logger_requests = logging.getLogger(__name__ + "::requestId")is if a user wanted to customize the strands logger now this won't fall under that unless there is a logger hierarchy system that I might not be aware of. But I'll explore this more

dbschmigelski committed:
feat(bedrock): add request ID logging via child logger

Add request ID logging for Bedrock API calls using a dedicated child
logger (strands.models.bedrock.requestId). This allows users to
enable request ID logging via Python's standard logging configuration
without requiring SDK-specific options.

- Log request ID at DEBUG level for successful requests
- Log request ID at INFO level for failed requests (exceptions)
- Child logger inherits from parent, respecting existing log config

zastrowm commented on tests/strands/models/test_bedrock.py:
```diff
@@ -1539,6 +1539,108 @@ async def test_stream_logging(bedrock_client, model, messages, caplog, alist):
+@pytest.mark.asyncio
+async def test_request_id_logging_streaming(bedrock_client, model, messages, caplog, alist):
+    """Test that request ID is logged at DEBUG level for streaming requests."""
+    import logging
```
nit: make top level import
